### PR TITLE
Add DirectXMath wrap

### PR DESCRIPTION
**Pull request by seungha-yang**
Adding DirectXMath (header only library) wrap.
See also https://github.com/microsoft/DirectXMath

### CODE REVIEW

File: subprojects/packagefiles/directxmath/meson.build
Comment (seungha-yang) [inline subprojects/packagefiles/directxmath/meson.build:5]: probably I should install headers with pkg-config file

File: subprojects/packagefiles/directxmath/meson.build
Comment (nirbheek) [inline subprojects/packagefiles/directxmath/meson.build:1]: Where is this version `3.1.8` coming from? The release tag says `3.18`: https://github.com/microsoft/DirectXMath/releases/tag/dec2022

Comment (seungha-yang) [reply subprojects/packagefiles/directxmath/meson.build:1]: uh, it seems to be inconsistent with version specified in cmake file then https://github.com/microsoft/DirectXMath/blob/dec2022/CMakeLists.txt#L6

File: subprojects/packagefiles/directxmath/meson.build
Comment (seungha-yang) [reply subprojects/packagefiles/directxmath/meson.build:18]: done

File: subprojects/packagefiles/directxmath/meson.build
Comment (seungha-yang) [reply subprojects/packagefiles/directxmath/meson.build:1]: I guess it might be mistake in their cmake file but I'd follow the cmake project version

Comment (seungha-yang) [reply subprojects/packagefiles/directxmath/meson.build:1]: Filed an issue https://github.com/microsoft/DirectXMath/issues/169

**Review by nirbheek — APPROVED** [review]
LGTM, but I'll wait 24h to see if someone else has comments before merging since this edits `tools/sanity_checks.py`.

**Pull request merged**
